src/containers/BooksForm.jsx

```diff
@@ -1,27 +1,80 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import { categories } from '../store';
+import { createBook } from '../actions';
 
-const BooksForm = () => (
-  <div>
-    <h3>Add New Book</h3>
-    <form>
-      <div>
-        <label htmlFor="title">
-          Title:
-          <input type="text" id="title" placeholder="Book Title" />
-        </label>
-      </div>
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    const { createBook } = this.props;
+    const { title, category } = this.state;
+    const book = {
+      id: Math.floor(Math.random() * 100),
+      title,
+      category,
+    };
+
+    if (title && category) {
+      createBook(book);
+      this.setState({
+        title: '',
+        category: '',
+      });
+    }
+
+    event.preventDefault();
+  }
+
+  render() {
+    const { title, category } = this.state;
+
+    return (
       <div>
-        <label htmlFor="category">
-          Category:
-          <select id="category">
-            <option selected>Select Category</option>
-            {categories.map(cat => <option key={cat} value={cat}>{cat}</option>)}
-          </select>
-        </label>
+        <h3>Add New Book</h3>
+        <form onSubmit={this.handleSubmit}>
+          <div>
+            <label htmlFor="title">
+              Title:
+              <input type="text" id="title" name="title" placeholder="Book Title" onChange={this.handleChange} value={title} />
```

Comment:
You could do the validation like so.
```suggestion
              <input type="text" id="title" name="title" placeholder="Book Title" onChange={this.handleChange} value={title} required />
```